server/server.js

```diff
@@ -1,34 +1,52 @@
+const bodyParser = require('body-parser');
 const express = require ('express');
 const server = express();
 const mongoose = require('mongoose');
 const User = require('./db/UserModel.js');
+const cors = require('cors');
+
+const STATUS_USER_ERROR = 422;
+const STATUS_SERVER_ERROR = 500;
+const STATUS_OKAY = 200;
+
+server.use(cors());
+server.use(bodyParser.json());
 
 
 server.get ('/', function (req, res) {
-    res.status(200).json ({message: 'API running'});
+   res.status(200).json ({message: 'API running'});
 });
 
-server.get ('/users', function (req, res) {
-   User.find({},(err, users)=> {
-       if (err) {
+server.get ('/api/users', function (req, res) {
+  User.find({},(err, users)=> {
+        if (err) {
+        res.status(STATUS_USER_ERROR).json({error:'Could not find the user.'});
+        } else {
+            res.status(200).json(users);
+        }
+  })
+});
 
-       } else {
-           res.status(200).json(users);
-       }
-   })
+server.post('/api/users/adduser', function(req, res) {
+    const newUser = new User (req.body);
+    //check the user contains all required data

```

Comment:
this works fine, if there is something missing the model will complain. 
something more explicit would be to destructure from the body and check if it's undefined. maybe you want to add that in as well as a double check , something like this 
{ email , username, password } = req.body;
if (!email || !username || !password) { 
res.status(400).json('please make sure all fields are filled in)}

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status